### DISCUSSION TITLE
Can pinocchio be used to identify robot dynamic parameters?

### DISCUSSION THREAD

**Discussion by LHT-Liu** [stack-of-tasks/pinocchio#2194] [Q&A]
Hello, team. Can pinocchio be used to identify robot dynamic parameters? Or do you know if there is any library related to dynamics identification?

**Comment by florent-lamiraux** [comment]
Hello @LIUHAITAO-CH .
You can try our toolbox based on pinocchio described here: https://laas.hal.science/hal-04234676v1/file/FIGAROH_IROS_2022.pdf.
The source code and installation instructions can be found here: https://gitlab.laas.fr/gepetto/figaroh.
If you  need some help, do not hesitate to contact the corresponding author of the paper.

**Reply by jcarpent** [reply]
Hi @LIUHAITAO-CH,

Depending on your needs, Pinocchio implements basic routines for doing robot identifications (these are typically leveraged in Figaroh). You can easily use them to implement your own identification pipeline.

**Reply by LHT-Liu** [reply]
ok，thanks，i‘ll study it

**Reply by cmastalli** [reply]
Our latest paper proposes a novel way to perform inertial estimation (and localization). This might be useful for anyone interested in this topic.

We have used Pinocchio to build this algorithm and we plan to open source it in Crocoddyl within the next months.

https://arxiv.org/pdf/2403.17161.pdf

**Reply by LHT-Liu** [reply]
> Our latest paper proposes a novel way to perform inertial estimation (and localization). This might be useful for anyone interested in this topic.
> 
> We have used Pinocchio to build this algorithm and we plan to open source it in Crocoddyl within the next months.
> 
> https://arxiv.org/pdf/2403.17161.pdf

thanks,where is code?
